web3/utils/abi.py

```diff
@@ -1,21 +1,512 @@
+        >>> abi = [
+        ...     {
+        ...         "constant": False,
+        ...         "inputs": [
+        ...             {"name": "a", "type": "uint256"},
+        ...             {"name": "b", "type": "uint256"},
+        ...         ],
+        ...         "name": "multiply",
+        ...         "outputs": [{"name": "result", "type": "uint256"}],
+        ...         "payable": False,
+        ...         "stateMutability": "nonpayable",
+        ...         "type": "function",
+        ...     }
+        ... ]
+        >>> fn_info = get_abi_element_info(abi, "multiply", [7, 3])
+        >>> fn_info["abi"]
+        {'constant': False, 'inputs': [{'name': 'a', 'type': 'uint256'}, {\
+'name': 'b', 'type': 'uint256'}], 'name': 'multiply', 'outputs': [{\
+'name': 'result', 'type': 'uint256'}], 'payable': False, \
+'stateMutability': 'nonpayable', 'type': 'function'}
+        >>> fn_info["selector"]
+        '0x165c4a16'
+        >>> fn_info["arguments"]
+        [7, 3]
+    """
+    args = args or tuple()
+    kwargs = kwargs or dict()
+
+    fn_abi = get_abi_element(
+        abi, function_identifier, args, kwargs, abi_codec=abi_codec
+    )
+    fn_selector = encode_hex(function_abi_to_4byte_selector(fn_abi))
+
+    if fn_abi["type"] == "fallback" or fn_abi["type"] == "receive":
+        return ABIElementInfo(abi=fn_abi, selector=fn_selector, arguments=tuple())
+    else:
+        fn_inputs = get_normalized_abi_inputs(fn_abi, args, kwargs)
+        _, aligned_fn_inputs = get_aligned_abi_inputs(fn_abi, fn_inputs)
+
+        return ABIElementInfo(
+            abi=fn_abi, selector=fn_selector, arguments=aligned_fn_inputs
+        )
+
+
+def get_abi_element(
+    abi: ABI,
+    function_identifier: FunctionIdentifier,
+    args: Optional[Sequence[Any]] = None,
+    kwargs: Optional[Any] = None,
+    abi_codec: Optional[Any] = None,
+) -> ABIElement:
+    """
+    Return the interface for an ``ABIElement`` which matches the provided identifier
+    and arguments.
+
+    The ABI which matches the provided identifier, named arguments (``args``) and
+    keyword args (``kwargs``) will be returned.
+
+    The `abi_codec` may be overridden if custom encoding and decoding is required. The
+    default is used if no codec is provided. More details about customizations are in
+    the `eth-abi Codecs Doc <https://eth-abi.readthedocs.io/en/latest/codecs.html>`__.
+
+    :param abi: Contract ABI.
+    :type abi: `ABI`
+    :param function_identifier: Find a function ABI with matching name.
+    :type function_identifier: `FunctionIdentifier`
+    :param args: Find a function ABI with matching args.
+    :type args: `list[Any]`
+    :param kwargs: Find a function ABI with matching kwargs.
+    :type kwargs: `Any`
+    :param abi_codec: Codec used for encoding and decoding. Default with \
+    `strict_bytes_type_checking` enabled.
+    :type abi_codec: `Any`
+    :return: ABI for the function interface.
```

Comment:
Not sure here but it's not only function, correct?
```suggestion
    :return: ABI for the specific ABI element.
```